src/snowflake/sqlalchemy/snowdialect.py

```diff
@@ -345,9 +368,46 @@ def get_pk_constraint(self, connection, table_name, schema=None, **kw):
         full_schema_name = self._denormalize_quote_join(
             current_database, schema if schema else current_schema
         )
-        return self._get_schema_primary_keys(
-            connection, self.denormalize_name(full_schema_name), **kw
-        ).get(table_name, {"constrained_columns": [], "name": None})
+
+        if table_name is not None:
+            return self._get_table_primary_keys(
+                connection,
+                self.denormalize_name(full_schema_name),
+                self.denormalize_name(table_name),
+                **kw,
+            ).get(table_name, {"constrained_columns": [], "name": None})
+        else:
+            return self._get_schema_primary_keys(
+                connection, self.denormalize_name(full_schema_name), **kw
+            ).get(table_name, {"constrained_columns": [], "name": None})
+
+    @reflection.cache
+    def _get_table_unique_constraints(self, connection, schema, table_name, **kw):
```

Comment:
same comment as the `_get_table_primary_keys`, let's avoid duplicating the code and reuse `_get_schema_unique_constraints`